src/core/p5.Renderer2D.js

```diff
@@ -155,10 +155,10 @@ p5.Renderer2D.prototype.image = function(
   }
 
   try {
+    if (p5.MediaElement && img instanceof p5.MediaElement) {
+      img.loadPixels();
```

Comment:
Actually, for the video freezing issue, what caused the problem was that when `get()` was called, the `canvas` property was set on the element with the pixel values at that point of time. Now in the `image` function, `cnv` got set to `img.canvas` instead of `img.elt` after calling `get()`. Because of this, the same old pixel values got rendered instead of updated values. This problem is not specific to `get()`. Any function that sets `canvas` property can cause the this issue because that `canvas` property does not get updated.

One solution could be to only load pixels on `MediaElement` if `canvas` property is set. That could better performance in several cases. But then we need to add extra check inside the `if(this._tint)` block to call `loadPixel` in case the element is `MediaElement`.  Shall I do this?

Reply:
I see... so if I understand correctly

* `loadPixels` is necessary for pixel operation (e.g. `get`) but you need to create a canvas for it, load pixels on it which can drag the performance
* for `image` and `copy` we actually don't need a canvas, so in fact we can skip `loadPixels`

and the current implementation is to **1)** "make sure" pixels are always loaded - which costs the performance (sometimes really badly) and **2)** there are mixups of parameters like in this case.

I would say, if the current PR fixes the issues and not breaking other stuff, we should merge it. And then we should open an issue (if not opened yet) about potentially cleaning up the back buffer canvas - whether they should be always loaded or only create a canvas when necessary. What do you think @outofambit ?

Reply:
hey @micuat this is the part of the library i'm least familiar with, but i can look into this soon. in the meantime, @stalgiag @aferriss do you happen to have any thoughts here?